05/GeoffRiley/similar_tweeters.py

```diff
@@ -0,0 +1,63 @@
+"""
+PyBites challenge 05: https://codechalleng.es/challenges/5/
+"""
+import re
+import string
+import sys
+
+from gensim import corpora, models, similarities
+import nltk
+from nltk.corpus import stopwords
+from user_tweets import UserTweets
+
+nltk.download('stopwords')
+sw = set(stopwords.words('english'))
+
+# exclusions:
+#  1. Tags starting with '#'
+#  2. email addresses containing '@'
+#  3. web addresses containing 'http' or 'https'
+#  4. URL indicators beginning '//'
+#  5. numbers comprising one or more digits followed by an optional two alphas
+exclusions = re.compile(r'^(#.*|.*@.*|.*https?|//.*|\d+([a-zA-Z]{2})?)$')
+my_punct = ''.join(x for x in string.punctuation if x not in ['#', '@'])
+trans = str.maketrans(my_punct, ' ' * len(my_punct))
+
+
+def clean_words(usr: UserTweets) -> list:
+    result = []
+    for twitter in usr:
+        for word in twitter.text.translate(trans).lower().split():
+            if word not in sw and len(word) > 3 and not exclusions.match(word):
```

Comment:
got a bit confused where `sw` was defined, I would name it `STOPWORDS` = more descriptive and uppercase makes me go to the top of the module

Reply:
you could make `3` a constant as well: `MIN_CHARS_WORD` or is that overkill?

Reply:
would this exclude the #AI hashtag btw?

Reply:
Yes, it is killing all hash tags… I don't really know how *significant* they are as part of a dialog.